### ISSUE TITLE
LQR shape bug

### ISSUE THREAD

**Issue by Lx017** [pypose/pypose#270]
### 🐛 Describe the bug

When using LQR, I got a shape error if **B** is of shape [n,m,1], that is for u being only a vector of one scalar.

The error raises at lqr.py line 340:
`F = torch.cat((A, B), dim=-1)`

I tried to fix it by changing line 338 and 339 from

`A = self.system.A.squeeze(-2)`
`B = self.system.B.squeeze(-1)`

to
`A = self.system.A.squeeze(-2) if self.system.A.ndim == 4 else self.system.A`
`B = self.system.B.squeeze(-1) if self.system.B.ndim == 4 else self.system.B`

Then I tested the modified code with example from https://pypose.org/docs/main/generated/pypose.module.LQR, test_mpc.py, and test_lqr.py and it worked fine. It is reproducible by changing **n_ctrl** to **1** in the first example code.

### Versions

PyTorch version: 2.0.1+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.2 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.27.0
Libc version: glibc-2.35

Python version: 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-5.19.0-46-generic-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.5
[pip3] torch==2.0.1
[pip3] torchaudio==2.0.2
[pip3] torchvision==0.15.2
[conda] Could not collect

**Comment by wang-chen** [comment]
This was fixed by 6421ae2e33121185083971b16398d8d8e1bbf39d
